### PR TITLE
Allow to add inject testing repository in the qt_compat test

### PR DESCRIPTION
**Pull request by bhush9**

### CODE REVIEW

**Comment by bhush9** [comment]
(don't merge yet... testing first)

File: bin/contain.rb [hunk 1]
```diff
@@ -67,6 +67,7 @@ def self.volume_specification_check(str)
 
 # Whitelist
 ENV_VARS = %w[
+  ADD_TESTING_REPO
```
Comment (hsitter) [inline bin/contain.rb]: plz no https://github.com/apachelogger/kde-os-autoinst/commit/7e796bf221e5c475e8566dbe4b51f3e3624318b3

**Review by hsitter — CHANGES_REQUESTED** [review]

**Comment by bhush9** [comment]
Okay runs now..

File: neon/tests/natives/qt_compat.pm [hunk 2]
```diff
@@ -61,6 +61,10 @@ sub run {
 
     # install packages
     select_console 'log-console';
+    if ( testapi::get_var("ADD_TESTING_REPO") ) {
```
Comment (hsitter) [inline neon/tests/natives/qt_compat.pm]: oh, excess spacing here. and you don't need to testapi:: here, note the `use testapi` above

File: bin/contain.rb (hunk 1 above)
Comment (bhush9) [reply bin/contain.rb]: fixed

File: neon/tests/natives/qt_compat.pm (hunk 2 above)
Comment (bhush9) [reply neon/tests/natives/qt_compat.pm]: fixed too..

File: neon/data/add_testing_repo.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+File.write('/etc/apt/preferences.d/neon-testing', <<-PREFERENCE)
+Package: *
+Pin: release l=Neon - Testing
+Pin-Priority: 1001
+PREFERENCE
+
+system('add-apt-repository "deb https://archive.neon.kde.org/testing xenial main"') || raise
+system('apt update') || raise
+system('apt dist-upgrade -y --allow-downgrades') || raise
```
Comment (bhush9) [inline neon/data/add_testing_repo.rb]: oh wait.. need to kill downgrades cli option

File: neon/data/add_testing_repo.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+ENV['DEBIAN_FRONTEND'] = 'noninteractive'
+
+File.write('/etc/apt/preferences.d/neon-testing', <<-PREFERENCE)
+Package: *
+Pin: release l=Neon - Testing
+Pin-Priority: 1001
+PREFERENCE
+
+system('add-apt-repository "deb https://archive.neon.kde.org/testing xenial main"') || raise
```
Comment (hsitter) [inline neon/data/add_testing_repo.rb:33]: not sure the hardcoding of xenial is particularly lovely given xenial will be dead in 2 months or so

**Pull request merged**